mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -363,6 +363,28 @@ static void atomicStore(OpBuilder &builder, Location loc,
   builder.create<memref::AtomicYieldOp>(loc, scalarMaskedValue);
 }
 
+/// Generate a non-atomic read-modify-write sequence for subbyte storing.
+/// It has similar logic to `atomicRMWStore`, but without atomicity.
```

Comment:
I don't have an answer yet, but we may need to replace `sub-byte` with `emulatedType` - not all "emulated" types within this file are sub-bytes. We also use e.g. `i32` to emulate `i8`:
https://github.com/llvm/llvm-project/blob/978310f1de712fc9092142f0985a541a537f21ec/mlir/test/Dialect/Vector/vector-emulate-narrow-type.mlir#L8-L27

Either we refrain from using sub-byte or force this logic to only be available for sub-bytes.